text/0000-cortex-m-targets.md

```diff
@@ -0,0 +1,608 @@
++    "cpu": "cortex-m3",
++    "llvm-target": "thumbv7m-none-eabi",
+ }
+```
+
+### cortex-m4
+
+``` diff
+     "target-pointer-width": "32",
++    "cpu": "cortex-m4",
++    "features": "+soft-float",
++    "llvm-target": "thumbv7em-none-eabi",
+ }
+```
+
+### cortex-m4f
+
+``` diff
+     "target-pointer-width": "32",
++    "cpu": "cortex-m4",
++    "llvm-target": "thumbv7em-none-eabi",
+ }
+```
+
+### cortex-m7
+
+``` diff
+     "target-pointer-width": "32",
++    "cpu": "cortex-m7",
++    "features": "+soft-float",
++    "llvm-target": "thumbv7em-none-eabi",
+ }
+```
+
+### cortex-m7f
+
+``` diff
+     "target-pointer-width": "32",
++    "cpu": "cortex-m7",
++    "llvm-target": "thumbv7em-none-eabi",
+ }
+```
+
+### cortex-m7f-sp
+
+``` diff
+     "target-pointer-width": "32",
++    "cpu": "cortex-m7",
++    "features": "+fp-only-sp",
++    "llvm-target": "thumbv7em-none-eabi",
+ }
+```
+
+### Rationale
+
+Behind the election of these values for the fields of the target specifications:
+
+- `arch`. (Used for conditional compilation). Set to `arm` to match the other ARM targets.
+- `data-layout`. (Code generation related). Also matches the value of the other ARM targets.
+- `executables`. Set to `true` because the Rust compiler can generate both libraries and executables
+  for these targets.
+- `os`. (For conditional compilation). No Rust precedent for this value as all the targets currently
+  supported by Rust have some form of OS or runtime. `none` has been chosen as this follows the
+  existing tooling (gcc and LLVM) convention of using `none` for the OS field of triples like
+  `arm-none-eabi`.
+- `pre-link-args` and `post-link-args`. By default, linkers are sensitive to the order of the
+  arguments they receive. Using these linker arguments undoes that default and can help avoid
+  "undefined reference" problems when linking statically. This technique is already used in the MUSL
+  targets that do static linking (`i686` and `x86_64` variants).
+- `target-endian`. (For conditional compilation). The Cortex-M architecture can be either little or
+  big endian. However, for microcontrollers, endianness is chosen at manufacture time and, as far as
+  the author knows, most (or all?) devices are little endian.
+- `target-pointer-width`. (For conditional compilation). All the existing Cortex-M processors are
+  32-bit processors.
+- `cpu`. (Enables CPU specific optimizations). The value pretty much matches the target name.
+- `features`. (Tweaks further the codege). On the Cortex-M4 and Cortex-M7 targets this tweaks the
+  use of FPU instructions. ARMv6-M based microcontrollers use `+strict-align` to prevent the
+  mis-optimization of the `copy_nonoverlapping` function.
+- `llvm-target`. (Controls code generation). Meaning of each field of the triple:
+  - `thumbv*m`. `thumb` indicates that the targets will use the Thumb instruction set instead of the
+    (older) ARM instruction set ("These processors support the Thumb instruction set only. -
```

Comment:
Nitpick: A64 supports 64-bit registers and pointers, but the actual size of the instructions is fixed 32-bit, like A32 (and unlike Thumb).